### PR TITLE
Updated Laravel Docs

### LINKED ISSUE #6789: Laravel Docs for Package AutoDiscovery
The Laravel Pacakge for Auth0 (https://github.com/auth0/laravel-auth0) is updated for package auto-discovery but the documentation hasn't been updated. The docs need to specify that the Service Provider and Facade are auto-discovered for Laravel 5.5+ and user is not required to add them manually.

### PR DESCRIPTION
**Pull request by gr8shivam**
Updated Docs to include information about Laravel Package Auto-Discovery.

Fixes #6789

### CODE REVIEW

**Comment by albertoperdomo** [comment]
If this PR is worth merging, do we need to also update the sample @joshcanhelp ?

**Comment by gr8shivam** [comment]
Updated samples with https://github.com/auth0-samples/auth0-laravel-php-web-app/pull/31

**Comment by joshcanhelp** [comment]
@gr8shivam - I left a comment on the sample PR:

https://github.com/auth0-samples/auth0-laravel-php-web-app/pull/31#pullrequestreview-166257452

TL;DR: Not working on my end with the steps we have there. Might be a minor documentation issue but I did not dig very far.

**Comment by gr8shivam** [comment]
Closed PR on sample since updating sample is not required. This PR can be merged directly.

**Comment by albertoperdomo** [comment]
>Closed PR on sample since updating sample is not required

Disagree. Tutorial and sample app should be updated together.
They are expected to work in the same way.

I see two options here:
* investigate further to get it running in the sample, merge both changesets
* close this PR

Are you willing to investigate further @gr8shivam ?
As in making sure the sample with the changeset runs as expected on a clean install.

**Comment by gr8shivam** [comment]
@albertoperdomo 

> They are expected to work in the same way.

That makes it easier then, simply remove `auth0/login` from `composer.json` in sample and integrate the Auth0 plugin after Laravel install. This will make sample in-line with the doc (https://auth0.com/docs/quickstart/webapp/laravel/01-login#install-the-auth0-plugin-and-its-dependencies)

Reopened PR (https://github.com/auth0-samples/auth0-laravel-php-web-app/pull/31#issuecomment-432695032) with suggested changes.

**Review by joshcanhelp — APPROVED** [review]

**Comment by cocojoe** [comment]
@joshcanhelp you approved, are you going to merge?

**Comment by joshcanhelp** [comment]
Will test and address in an upcoming Laravel major.
